### DISCUSSION TITLE
preserverdValues for preserving user input but not on the submitted data

### DISCUSSION THREAD

**Discussion by ha1fstack** [TanStack/form#708] [Ideas] [answered]
I have a field on the form that's conditionally rendered which references other fields on the form.

When the field is not visible I don't need it on the submission result.
If I set the `preserveValue` to false for the field - which is default behavior, it works as expected.

But for user experience, I want to turn on `preserveValue` so the user input values can be persisted even when the field is not visible.
However turning on `preserveValue` validates the field on submit even when it's not rendered and the field value is not needed, so if the field input is not valid the submission would fail even when the field is not rendered.

It would be great to have temporary preservedValues option that only saves the value on the user side but the input is not validated or included in the submission result if the field is not rendered.

**Comment by crutchcorn** [comment, marked as answer]
We removed this API as it didn't make sense to us and was intensely buggy with StrictMode.

What was your usecase for the API?

**Reply by ha1fstack** [reply]
Thank you for attention! As I wrote, 

> But for user experience, I want to turn on preserveValue so the user input values can be persisted even when the field is not visible.

for this reason.

**Reply by crutchcorn** [reply]
Oh! Sorry I missed that.
